### PR TITLE
OBSDOCS-1020/TRACING-4075: Write documentation for the k8sobjectsreceiver component

### PR DESCRIPTION
**Pull request by max-cx**
:warning: This PR is tied to a product release date, which is currently set to **June 5** (Wednesday).

- If the merge review gets delayed, you're welcome to ping `mleonov` on Slack.
- If you have any questions, ping `mleonov` on Slack.

Signed-off-by: Benedikt Bongartz <bongartz@klimlive.de>

Edit of https://github.com/openshift/openshift-docs/pull/74966

Version(s):

4.12, 4.13, 4.14, 4.15, 4.16

Issue:

https://issues.redhat.com/browse/OBSDOCS-1020
https://issues.redhat.com/browse/TRACING-4076

Link to docs preview:

https://75335--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-configuration-of-otel-collector.html#k8sobjectsreceiver-receiver_otel-configuration-of-otel-collector

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by max-cx** [comment]
@IshwarKanse, I'll push this PR for a peer review, and we can add the missing info in a separate PR.

**Review by GroceryBoyJr — COMMENTED** [review]
LGTM

**Comment by max-cx** [comment]
Closing this PR because I cherry-picked its commit into https://github.com/openshift/openshift-docs/pull/76340.

File: modules/otel-collector-components.adoc [hunk 1]
```diff
@@ -82,6 +82,42 @@ The Jaeger receiver ingests traces in the Jaeger formats.
...
+          mode: watch # <2>
+          interval: # <3>
+          label_selector: # <4>
+          field_selector: # <5>
+          namespaces: [<namespace>,...] # <6>
+    service:
+      pipelines:
+        logs:
+          receivers: [k8sobjects]
+
```
Comment (frzifus) [inline modules/otel-collector-components.adoc:112]:
```suggestion
apiVersion: v1
kind: ServiceAccount
metadata:
  name: otel-k8sobj
  namespace: <namespace>
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: otel-k8sobj
  namespace: <namespace>
rules:
- apiGroups:
  - ""
  resources:
  - events
  - pods
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - "events.k8s.io"
  resources:
  - events
  verbs:
  - watch
  - list
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: otel-k8sobj
subjects:
  - kind: ServiceAccount
    name: otel-k8sobj
    namespace: <namespace>
roleRef:
  kind: ClusterRole
  name: otel-k8sobj
  apiGroup: rbac.authorization.k8s.io
---
apiVersion: opentelemetry.io/v1alpha1
kind: OpenTelemetryCollector
metadata:
  name: otel-k8s-obj
  namespace: <namespace>
spec:
  serviceAccount: otel-k8sobj
  image: ghcr.io/os-observability/redhat-opentelemetry-collector/redhat-opentelemetry-collector:main
  mode: deployment
  config: |
    receivers:
      k8sobjects:
        auth_type: serviceAccount
        objects:
          - name: pods # <1>
            mode: pull # <2>
            interval: 30s # <3>
            label_selector: # <4>
            field_selector: # <5>
            namespaces: [<namespace>,...] # <6>
          - name: events
            mode: watch
    exporters:
      debug:
    service:
      pipelines:
        logs:
          receivers: [k8sobjects]
          exporters: [debug]
```

**Pull request merged**